components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -779,33 +779,34 @@ void RewardsServiceImpl::TriggerOnWalletInitialized(int error_code) {
 
 void RewardsServiceImpl::TriggerOnWalletProperties(int error_code,
     std::unique_ptr<ledger::WalletInfo> wallet_info) {
-  std::unique_ptr<brave_rewards::WalletProperties> wallet_properties;
+  for (auto& observer : observers_) {
+    std::unique_ptr<brave_rewards::WalletProperties> wallet_properties;
 
-  if (!wallet_info) {
-    return;
-  }
+    if (!wallet_info) {
+      return;
```

Comment:
this should still be outside the loop

Reply:
actually that isn't quite right either

Reply:
this block should actually be removed and reverted back to the original version here https://github.com/brave/brave-core/commit/54199c354eed1fcbf6f3f429f1cdeafb405b0369#diff-a5d16be9c9b52a6df7b866b1d75d3f5fL784

Reply:
we always want OnWalletProperties to fire even in cases where wallet_info is null because `!wallet_info` always means that an error occurred and the observers should be notified so they can take any relevant action cc @NejcZdovc